clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala

```diff
@@ -117,12 +118,12 @@ trait GvcfTests {
       } yield {
         outputs should be(Seq(expected))
 
-        val storedDocument = getJsonFrom[DocumentGvcf](returnedUpsertId)
-        storedDocument.location should be(expected.location)
-        storedDocument.project should be(expected.project)
-        storedDocument.sampleAlias should be(expected.sampleAlias)
-        storedDocument.version should be(expected.version)
-        storedDocument.gvcfPath should be(expected.gvcfPath)
+        val storedDocument = getJsonFrom(returnedUpsertId)(ElasticsearchIndex.Gvcf)
+        getLocation(storedDocument) should be(expected.location)
+        getStringByName(storedDocument, "project") should be(expected.project)
+        getStringByName(storedDocument, "sampleAlias") should be(expected.sampleAlias)
+        getIntByName(storedDocument, "version") should be(expected.version)
+        getUriByName(storedDocument, "gvcfPath") should be(expected.gvcfPath)
```

Comment:
If you do keep these `get` methods, you should be able to collapse them all into one generic:
`def getByName[A : Decoder](doc: Json, name: String): A`

Reply:
Tried that - compiler doesn't like not knowing the type ahead of time.

Reply:
Hwot! This should definitely be doable... what's the error you hit if you try it?